### PR TITLE
Share logic for asynchronous connect in AbstractChannel

### PR DESCRIPTION
**Pull request by normanmaurer**
Motivation:

We had a lot of duplication to handle asynchronous connect in multiple implementations. Let's share it and so remove a lot of duplication.

Modifications:

- Move logic into AbstractChannel
- Add abstract methods doConnect(...) and doFinishConnect(...)
- Add isConnectPending()

Result:

Share code / cleanup and better abstract base class

### CODE REVIEW

**Comment by normanmaurer** [comment]
There is still some race … looking into it

**Comment by normanmaurer** [comment]
@chrisvest this is ready now

**Comment by normanmaurer** [comment]
will fix the conflict before merging (once approved)

File: transport/src/test/java/io/netty5/channel/AbstractChannelTest.java [hunk 1]
```diff
@@ -230,11 +228,14 @@ public ChannelMetadata metadata() {
...
-        protected void connectTransport(
-                SocketAddress remoteAddress, SocketAddress localAddress, Promise<Void> promise) {
-            promise.setFailure(new UnsupportedOperationException());
+        protected boolean doConnect(SocketAddress remoteAddress, SocketAddress localAddress) {
+            throw new UnsupportedOperationException();
         }
 
+        @Override
+        protected boolean doFinishConnect(SocketAddress requestedRemoteAddress) {
+            throw new UnsupportedOperationException();        }
```
Comment (chrisvest) [inline transport/src/test/java/io/netty5/channel/AbstractChannelTest.java:237]: Got some messed up formatting here.

File: transport/src/test/java/io/netty5/channel/DefaultChannelPipelineTailTest.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
  */
 package io.netty5.channel;
 
-import io.netty5.util.concurrent.Promise;
+
```
Comment (chrisvest) [inline transport/src/test/java/io/netty5/channel/DefaultChannelPipelineTailTest.java:18]: Extra line break?

**Pull request merged**